validator/client/propose.go

```diff
@@ -257,3 +267,47 @@ func signVoluntaryExit(
 	}
 	return sig.Marshal(), nil
 }
+
+// graffiti gets the graffiti from cli or file for the validator public key.
+func (v *validator) getGraffiti(ctx context.Context, pubKey [48]byte) ([]byte, error) {
+	// When specified, default graffiti from the command line takes the first priority.
+	if len(v.graffiti) != 0 {
+		return v.graffiti, nil
+	}
+
+	if v.graffitiStruct == nil {
+		return nil, errors.New("graffitiStruct can't be nil")
+	}
+
+	// When specified, default graffiti from the file takes the second priority.
+	if v.graffitiStruct.Default != "" {
+		return []byte(v.graffitiStruct.Default), nil
+	}
+
+	// When specified, a graffiti from the sequential list in the file will take third priority.
+	if len(v.graffitiStruct.Sequential) != 0 {
+		g := []byte(v.graffitiStruct.Sequential[0])
+		v.graffitiStruct.Sequential = v.graffitiStruct.Sequential[1:]
+		return g, nil
+	}
+
+	// When specified, a graffiti from the random list in the file take fourth priority.
+	if len(v.graffitiStruct.Random) != 0 {
+		r := rand.NewGenerator()
+		r.Seed(time.Now().Unix())
+		i := r.Uint64() % uint64(len(v.graffitiStruct.Random))
+		return []byte(v.graffitiStruct.Random[i]), nil
+	}
+
+	// Last and when specified, individual validator specified graffiti will be used.
+	idx, err := v.validatorClient.ValidatorIndex(ctx, &ethpb.ValidatorIndexRequest{PublicKey: pubKey[:]})
+	if err != nil {
+		return []byte{}, err
+	}
```

Comment:
Can this be cached some how? We know this information from the duties as well

Reply:
Discussed offline, there's no cache but this is ok. The call is cheap and proposing block is not a frequent event